exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/utils/model_retrieval.ipynb

```diff
@@ -7,12 +7,12 @@
    "source": [
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #4.        There must be a global variable - sb_config - providing access to the secret store holding
</pre><p>What is the reason for this change?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/exasol/data-science-sandbox/pull/119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Reply:
Just made it uniform across all notebooks. There is a universal assumption that sb_config exists. Passing it as a parameter in this case is a bit confusing.

Reply:
long term, I would say at least the implementations of the methods should move to notebook-connector, that means they would need the SecretStore as a parameter, in general using global variables in function calls for trouble

Reply:
The implementation is already in the notebook-connection. I agree that a global variable is generally not a good idea. But this is one exception that we use everywhere. Trying to irradicate it now would be impractical.

Reply:
More to the point, the Jupyter programming model is a bit different from the normal Python. The cells share data via global variables. There is no encapsulation. I.e., it's not like in Excel where you address a content of the cell through an object model, for examples `range("B3")`.

Reply:
I know that it has a different model, that doesn't we should use it, because in my opinion, it encourages bad practice. In general, I would say the following practice is in Jupyter the advisable. Use the shared state between cells only in the cells, but not in your own functions. Means, you can define a variable in one cell and use it in another, but if you call functions, they always get their input via parameters.

Reply:
However, I can live for the moment with the global variable, but please create a ticket for its removal.

Reply:
OK, deal. [Here](https://github.com/exasol/data-science-sandbox/issues/120) is the ticket.